ledger/eval.go

```diff
@@ -1049,6 +1084,47 @@ func (eval *BlockEvaluator) endOfBlock() error {
 	return nil
 }
 
+// modifyOfflineAccounts after all transactions and rewards are processed, modify the accounts so that their status is offline
+func (eval *BlockEvaluator) modifyOfflineAccounts() error {
+
+	proto, ok := config.Consensus[eval.block.BlockHeader.CurrentProtocol]
+	if !ok {
+		return fmt.Errorf("unknown consensus version: %s", eval.block.BlockHeader.CurrentProtocol)
+	}
+
+	expectedMaxNumberOfExpiredAccounts := proto.MaxExpiredAccountsToProcess
+	lengthOfExpiredParticipationAccounts := len(eval.block.ParticipationUpdates.ExpiredParticipationAccounts)
+
+	// If the length of the array is strictly greater than our max then we have an error.
+	// This works when the expected number of accounts is zero (i.e. it is disabled) as well
+	if lengthOfExpiredParticipationAccounts > expectedMaxNumberOfExpiredAccounts {
+		return fmt.Errorf("length of expired accounts (%d) was greater than expected (%d)",
+			lengthOfExpiredParticipationAccounts, expectedMaxNumberOfExpiredAccounts)
+	}
+
+	for _, accountAddr := range eval.block.ParticipationUpdates.ExpiredParticipationAccounts {
+		acctData, err := eval.state.lookup(accountAddr)
+		if err != nil {
+			return fmt.Errorf("modifyOfflineAccounts wrong: %v was not found", accountAddr)
```

Comment:
an error would be returned when the lookup failed, but not when the lookup was looking for a non-existing account.
If you were to look a non-existing account, you would get the "empty" account data.
Therefore, returning an error here is good - but the error string isn't; it should say something like:
```golang
    return fmt.Errorf("modifyOfflineAccounts was unable to retrieve account %v : %w", accountAddr, err)
```